lib/ansible/module_utils/k8s/raw.py

```diff
@@ -210,18 +210,19 @@ def perform_action(self, resource, definition):
             if self.check_mode:
                 k8s_obj = dict_merge(existing.to_dict(), definition)
             else:
-                if self.params['merge_type']:
-                    from distutils.version import LooseVersion
-                    if LooseVersion(self.openshift_version) < LooseVersion("0.6.2"):
+                from distutils.version import LooseVersion
+                if LooseVersion(self.openshift_version) < LooseVersion("0.6.2"):
+                    if self.params['merge_type']:
                         self.fail_json(msg="openshift >= 0.6.2 is required for merge_type")
-                    for merge_type in self.params['merge_type']:
+                    else:
+                        k8s_obj, error = self.patch_resource(resource, definition, existing, name,
+                                                             namespace)
+                else:
+                    for merge_type in self.params['merge_type'] or ['strategic-merge', 'merge']:
```

Comment:
Note: It's nicer code organization to do parameter validation and normalization upfront rather than at usage.  So it would be better to move this version checking and default setting to somewhere soon after the KubernetesAnsibleModule is created.